lib/Mojo/Util.pm

```diff
@@ -88,6 +88,33 @@ monkey_patch(__PACKAGE__, 'sha1_sum',      \&sha1_hex);
 monkey_patch(__PACKAGE__, 'steady_time',
   MONOTONIC ? sub () { Time::HiRes::clock_gettime(Time::HiRes::CLOCK_MONOTONIC()) } : \&Time::HiRes::time);
 
+# Building random bytes function
+monkey_patch(__PACKAGE__, 'urandom_bytes', $^O eq 'MSWin32' ? _build_urandom_bytes_win32() : _build_urandom_bytes());
+
+sub _build_urandom_bytes_win32 {
+  require Win32::API;
+
+  my $RtlGenRandom = Win32::API->new('advapi32', 'INT SystemFunction036 (PVOID RandomBuffer, ULONG RandomBufferLength)')
+    or croak "SystemFunction036 import failed: $!";
+
+  return sub {
+    my $num   = shift || 64;
+    my $bytes = chr(0) x $num;
+    $RtlGenRandom->Call($bytes, $num) or croak "SystemFunction036 call failed: $!";
+    return $bytes;
+  }
+}
+
+sub _build_urandom_bytes {
+  open(my $urandom, '<', '/dev/urandom') or croak "Cannot open /dev/urandom: $!";
```

Comment:
I'm not sure if it's a good idea to leave a read handle open to this for the process life. But maybe it's fine. Would there be any issues with sharing this over forks or threads?

Reply:
- `urandom_bytes`: Afaik it should be ok with `sysread` to keep an open file handle across forks, but it now opens `/dev/urandom` on each invocation rather than
   keeping the fh around, as suggested by @Grinnz. This should be less of a problem now that `urandom_urlsafe` encodes 32 bytes directly. 
   
   The Win32::API `SystemFunction036` object is kept using `state`.